### ISSUE TITLE
Migration to Java 11 from Java 8

### ISSUE THREAD

**Issue by chenejac** [vivo-project/VIVO#3834]
**Is your feature request related to a problem? Please describe.**
Some libraries/dependences require JDK 11+, e.g. Jena 4.

**Describe the solution you'd like**
Migration to JDK 11 including upgrading some libraries/dependencies if that is needed. 

**Describe alternatives you've considered**
Migration to JDK 17

**Comment by michel-heon** [comment]
For many of my assignments it is better to work with the new version (11 or 17). 
Here is the list of tasks depending on it.
[Testing performance of external graph usage](https://github.com/vivo-project/VIVO/issues/3823)
[Improvement of communication with external graph](https://github.com/vivo-project/VIVO/issues/3824)
[Investigate and specify support for reasoning](https://github.com/vivo-project/VIVO/issues/3827)

and also,  all task associated to "Jena TDB"

**Comment by michel-heon** [comment]
from my point of view, as long as we have to do the work, I think it's better to go directly to Java 17 and thus avoid possible incompatibilities problems that could occur during a future transition from 11 to 17. I'm thinking in particular of the integration of the reasoner whose source code development has been stopped for a few years. It would be a bad thing to lose some developed functionalities because of an incompatibility between version 11 and 17 (or higher)

Translated with www.DeepL.com/Translator (free version)

**Comment by michel-heon** [comment]
I propose that this task be moved to "jena tdb".

**Comment by michel-heon** [comment]
@chenejac @wwelling @litvinovg 
I just found this document: [Oracle JDK Migration Guide](https://docs.oracle.com/en/java/javase/11/migrate/index.html)
Before going further in the PR's, I propose to review the work already done against this specification 
I haven't read the whole thing yet but I think there are some good recommendations on what to do when considering an upgrade to Java11

**Comment by chenejac** [comment]
It looks as jdeps might be useful in the migration process
https://docs.oracle.com/en/java/javase/11/tools/jdeps.html 
https://docs.oracle.com/en/java/javase/11/tools/jdeprscan.html

@michel-heon did you already try to run those checks?

**Comment by bkampe** [comment]
I'm using Java 11 since a while without any problems. I guess this ticket could be closed, couldn't it?

**Comment by hauschke** [comment]
Yes, seems so. 

https://wiki.lyrasis.org/spaces/VIVODOC116x/pages/364742269/System+Requirements#SystemRequirements-JavaJava8or11

Java 11 is already supported according to the documentation, too. Can be closed, @chenejac. Would be a good idea to create a new ticket for JDK 17 which references this ticket for some background information.

**Comment by chenejac** [comment]
Yes, this is resolved (https://github.com/vivo-project/Vitro/commit/95ef98c589931971da317c4daa86f3fc195ab038) and can be closed. We will create the new ticket for migration to Java 17
